### PR TITLE
Ray tracing doco improvements

### LINKED ISSUE #16507: RayTracing: doco suggestions
## Reason

I was just reading the doco and there are a few places where i thought it could be slightly modified.  Tagging @loganharbour .   Feel free to close if you think these suggestions stink.

## Design

- Does RayTracing only work for 2D meshes?  The lead page states "planar finite elements" that suggests 2D, but then in my naivety i can't see why 3D meshes are super difficult.
- In the first equation of https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html , what is "u_h"?  Why isn't it just "u"?   Also, i wouldn't include the measure (dr) as i think that's confusing.
- The second equations of https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html should have r_1 and r_2 - currently they just have r_1.
- I think it'd be nice in https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html  to actually write what you're trying to compute, viz something like \int_{0}^{5} dx sqrt{2}x/5 (or however you want to write it), just because in order to verify the results i had to look back at the input file and see that u = x/5 for this example.
- In https://mooseframework.inl.gov/modules/ray_tracing/examples/line_sources.html i really think you need to write the DE you're trying to solve, viz \nabla^{2}u = 5\delta(x-ray), where ray is specified by the RepeatableRayStudy (and \delta is the 2D Dirac delta function).  The reason is that it took me a while to figure this out - i wondered what the "5" was in LineSourceRayKernel, for instance.
- Might be nice to see the mesh in the pics in https://mooseframework.inl.gov/modules/ray_tracing/examples/line_sources.html 
- Might be nice to mention more about Jacobians.  These are really important in PorousFlow so i need to know whether i can include all Jacobian contributions.

## Impact
Better user experience with improved documentation.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes #16507

@WilkAndy, for starters, thank you so much for taking a look at this. I've been pretty busy working on getting the implementation documented which left more to be desired for the simple examples.

Here's a summary with your comments:

> Does RayTracing only work for 2D meshes? The lead page states "planar finite elements" that suggests 2D, but then in my naivety i can't see why 3D meshes are super difficult.

It definitely does. What I meant to say here was planar _sides_. We also support non-planar sides to some extent with an approximation, which does pretty well if your mesh is only a _little_ non-planar. 

See 2bdad2d56971c74dd8756e2575fddacdda62453a and https://mooseframework.inl.gov/docs/PRs/16512/site/modules/ray_tracing/index.html

> In the first equation of https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html , what is "u_h"? Why isn't it just "u"? Also, i wouldn't include the measure (dr) as i think that's confusing.

This is u_h because it's the integral of the FEM solution, not the true analytic solution. For the comment about dr, I changed u_h -> u_h(\vec{r}) to make this clearer.

See https://mooseframework.inl.gov/docs/PRs/16512/site/modules/ray_tracing/examples/line_integrals.html

> The second equations of https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html should have r_1 and r_2 - currently they just have r_1.

Fixed in 3df19a699a8685e8133bc5dcee929b9ee393322a.

> I think it'd be nice in https://mooseframework.inl.gov/modules/ray_tracing/examples/line_integrals.html to actually write what you're trying to compute, viz something like \int_{0}^{5} dx sqrt{2}x/5 (or however you want to write it), just because in order to verify the results i had to look back at the input file and see that u = x/5 for this example.

I'm a bit stuck on this because that would require me getting the analytic solution which I think is a little overkill for this example. I really do think that the general form is good enough, but if you disagree I can try at it.

> In https://mooseframework.inl.gov/modules/ray_tracing/examples/line_sources.html i really think you need to write the DE you're trying to solve, viz \nabla^{2}u = 5\delta(x-ray), where ray is specified by the RepeatableRayStudy (and \delta is the 2D Dirac delta function). The reason is that it took me a while to figure this out - i wondered what the "5" was in LineSourceRayKernel, for instance.

I agree. I added the strong form for this system in f69aef27794c84268259502b8fb759f2d987bde6 and https://mooseframework.inl.gov/docs/PRs/16512/site/modules/ray_tracing/examples/line_sources.html

> Might be nice to see the mesh in the pics in https://mooseframework.inl.gov/modules/ray_tracing/examples/line_sources.html

Done and done in 7bef522b36fbcc61dbb3d8351854a4b839c82150 and https://mooseframework.inl.gov/docs/PRs/16512/site/modules/ray_tracing/examples/line_sources.html

> Might be nice to mention more about Jacobians. These are really important in PorousFlow so i need to know whether i can include all Jacobian contributions.

This one is a toughie. We definitely support full coupling. The issue is finding an example that involves coupling and a line source that is general enough to put in the framework. There is a test object for this (see https://github.com/idaholab/moose/blob/next/modules/ray_tracing/test/src/ray_kernels/CoupledLineSourceRayKernelTestTempl.C). If you can think of an example, please let me know.

For now, I added a simple comment about full coupling support in 40b4cdbb87db1049b3d77919d5a1e59970db1995 and https://mooseframework.inl.gov/docs/PRs/16512/site/modules/ray_tracing/index.html.

### CODE REVIEW

**Comment by loganharbour** [comment]
Ignore the failing test - testing a new recipe on this PR.

File: modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md [hunk 1]
```diff
@@ -14,17 +14,17 @@ We begin with the standard "simple diffusion" problem:
 For this problem, we seek the value of the integral
 
 !equation
-\int_L u_h~dr, \quad L = \{\vec{r}_1 + t\vec{r}_2 \mid t \in [0, 1]\}
+\int_L u_h(\vec{r})~dr, \quad L = \{\vec{r}_1 + t\vec{r}_2 \mid t \in [0, 1]\}
```
Comment (WilkAndy) [inline modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md:17]: I understand now what you mean by u_h.  I think you should either:

- state "u_h is the finite-element solution for u"
- or simply have "u".  I hope that MOOSE users will understand that MOOSE results are subject to finite-element discretisation approximations, so won't expect their line integrals to always agree with the analytic solution.

File: modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md [hunk 2]
```diff
@@ -14,17 +14,17 @@ We begin with the standard "simple diffusion" problem:
...
-\vec{r}_1 = (0, 0)\,, \quad \vec{r}_1 = (5, 5)
+\vec{r}_1 = (0, 0)\,, \quad \vec{r}_2 = (5, 5)
 
 and
 
 !equation
-\vec{r}_1 = (5, 0)\,, \quad \vec{r}_1 = (5, 5)
+\vec{r}_1 = (5, 0)\,, \quad \vec{r}_2 = (5, 5)
 
 For simplicity, we will denote the lines as `diag` and `right_up`, respectively.
```
Comment (WilkAndy) [inline modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md:29]: It's possible that the Diffusion Kernel is confusing in this example (a reader might think "but isn't the Kernel going to change u?"), but i can't see a nice way of avoiding that.  Perhaps you could also include the statement that u(x, y, t) = x/5.

**Review by WilkAndy — DISMISSED** [review]
Looks great - just a couple of minor suggestions that you could address or choose to ignore - up to you.

File: modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md (hunk 2 above)
Comment (loganharbour) [reply modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md]: Let me know what you think about 7f0dfebcbf4e81857da35401e395da156b830cee:

Screen Shot 2020-12-16 at 3 47 00 PM

File: modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md (hunk 1 above)
Comment (loganharbour) [reply modules/ray_tracing/doc/content/modules/ray_tracing/examples/line_integrals.md]: 👍🏼

**Review by zachmprince — APPROVED** [review]

**Pull request merged**